ambry-store/src/test/java/com.github.ambry.store/StorageManagerTest.java

```diff
@@ -516,5 +573,39 @@ private void generateConfigs(boolean segmentedLog) {
     diskManagerConfig = new DiskManagerConfig(vProps);
     storeConfig = new StoreConfig(vProps);
   }
+
+  /**
+   * An extension of {@link ReplicaStatusDelegate} to help with tests.
+   */
+  private static class MockReplicaStatusDelegate extends ReplicaStatusDelegate {
```

Comment:
you should be able to mock this through `Mockito` instead of writing this full blown mock. Were there any problems with that?

Reply:
yeah. I will take your suggestion. Thanks.

Reply:
Hmm - may still be able to use Mockito. Let's chat offline once all other comments are addressed